search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/llm/DefaultLlmImpl.java

```diff
@@ -161,6 +175,19 @@ protected ChatCompletionOutput buildChatCompletionOutput(ModelProvider provider,
                     errors = List.of("Unknown error or response.");
                 }
             }
+        } else if (provider == ModelProvider.COHERE) {
+            String response = (String) dataAsMap.get("text");
+            if (response != null) {
+                answers = List.of(response);
+            } else {
+                Map error = (Map) dataAsMap.get("error");
+                if (error != null) {
+                    errors = List.of((String) error.get("message"));
+                } else {
+                    errors = List.of("Unknown error or response.");
+                    log.error("{}", dataAsMap);
+                }
```

Comment:
seems like this block is completely similar except this line: `String response = (String) dataAsMap.get("text");` for bedrock it's `String response = (String) dataAsMap.get("completion");`.

Reply:
Yeah, it's a bit inefficient.  I left it this way for now as I am not 100% how we'd converge on error handling across different models/providers.  I am focusing mostly on happy paths.  In the event of an error, this implementation does capture it (attempts to, at least).  But what I don't cover is all the different ways remote endpoints include errors in the responses.

Reply:
may be for now just put into a method expecting (text/completion) as parameter? At least that will reduce the lines :)

Reply:
are we addressing this?

Reply:
Yes, please check the latest diff.